src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Usage/UseCuriouslyRecurringTemplatePatternCorrectly.cs

```diff
@@ -0,0 +1,126 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Analyzer.Utilities;
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.NetCore.Analyzers.Usage
+{
+    using static MicrosoftNetCoreAnalyzersResources;
+
+    public abstract class UseCuriouslyRecurringTemplatePatternCorrectly : DiagnosticAnalyzer
```

Comment:
(nit) I think I would name this class and the rule more targeted. Something like `ImplementGenericMathInterfacesCorrectly`.